packages/carbon-graphs/src/js/controls/Graph/helpers/helpers.js

```diff
@@ -519,6 +529,16 @@ const updateXAxisDomain = (config, input = {}) => {
 
   config.axis.x.domain = padDomain(newDomain, config.axisPadding.x);
 
+  // if the x-axis type is timeseries then convert the updated epoc date back to a string
+  if (config.axis.x.type === AXIS_TYPE.TIME_SERIES) {
+    config.axis.x.domain.upperLimit = utils.getDateFromEpoc(config.axis.x.domain.upperLimit);
+    config.axis.x.domain.lowerLimit = utils.getDateFromEpoc(config.axis.x.domain.lowerLimit);
```

Comment:
After the upperLimit and lowerLimit are converted to epoc, do they get altered in anyway in the logic above? If they do get altered in any way, then ignore this comment. Otherwise, would it also work if we just created a temp constant to save off the upperLimit and lowerLimit before converting them to epoc and then just reset here so we don't need to convert them back to a date string?

Anyways, the way it is now looks good to me.

Reply:
They do get altered by getting updated to new values and the logic to get new values is integer-based, so basically, they are converted to epoc, updated then converted back.